clusterstate/clusterstate.go

```diff
@@ -0,0 +1,130 @@
+package clusterstate
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+
+	"github.com/giantswarm/e2etests/clusterstate/provider"
+)
+
+type Config struct {
+	Logger   micrologger.Logger
+	Provider provider.Interface
+}
+
+type ClusterState struct {
+	logger   micrologger.Logger
+	provider provider.Interface
+}
+
+func New(config Config) (*ClusterState, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+	if config.Provider == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Provider must not be empty", config)
+	}
+
+	s := &ClusterState{
+		logger:   config.Logger,
+		provider: config.Provider,
+	}
+
+	return s, nil
+}
+
+func (c *ClusterState) Test(ctx context.Context) error {
+	var err error
+
+	{
+		c.logger.LogCtx(ctx, "level", "debug", "message", "installing e2e-app")
+
+		err = c.provider.InstallTestApp()
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		c.logger.LogCtx(ctx, "level", "debug", "message", "installed e2e-app")
```

Comment:
We install the test app to make sure the cluster doesn't get wiped when rebooting or replacing the master.

Reply:
In the WaitForGuestReady I also check for 2 pods of the e2e app. Sounds like I should split that out.

Reply:
Done.